src/sagas/ui.js

```diff
@@ -12,26 +17,39 @@ export function* userDoneTyping() {
   yield put(userDoneTypingAction());
 }
 
-export function* exportGist() {
+function* _githubExport(
+  successAction,
+  failureAction,
+  notDisplayedAction,
+  displayedAction) {
   const exportWindow = yield call(
     openWindowWithWorkaroundForChromeClosingBug,
     `data:text/html;charset=utf-8;base64,${spinnerPage}`,
   );
   const {type, payload: url} =
-    yield take(['GIST_EXPORTED', 'GIST_EXPORT_ERROR']);
+    yield take([successAction, failureAction]);
 
-  if (type === 'GIST_EXPORTED') {
+  if (type === successAction) {
     if (exportWindow.closed) {
-      yield put(gistExportNotDisplayed(url));
+      yield put(notDisplayedAction(url));
     } else {
       exportWindow.location.href = url;
-      yield put(gistExportDisplayed());
+      yield put(displayedAction());
     }
   } else {
     yield call([exportWindow, 'close']);
   }
 }
 
+export function exportGist() {
```

Comment:
I don't think so...I originally had these as generators which yielded githubExport, but the spinner didn't render right and the tests failed because there was an extra `.next()` needed to call - oddly enough the gist/repo still got exported properly at the end, don't really know enough about sagas/generators to know why that was. Anyway, this function and `exportRepo` instead return a generator, instead of being a generator themself